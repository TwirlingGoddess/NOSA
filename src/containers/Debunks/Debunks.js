import React, { Component } from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';

export class Debunks extends Component {
  render() {
    const newImages = this.props.debunks.map(image => <img className="yessa" alt="" src={image.url} />);
    return (
      <div className="Pics">
        <section>{newImages}</section>
        <NavLink to="/prints">Print Image</NavLink>
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  debunks: state.debunks,
});

Debunks.propTypes = {
  debunks: PropTypes.array,
};

export default connect(mapStateToProps, null)(Debunks);
