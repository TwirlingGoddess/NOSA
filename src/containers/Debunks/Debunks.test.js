import React from 'react';
import { Debunks, mapStateToProps } from './Debunks';
import { shallow } from 'enzyme';


describe('Debunks container', () => {
  let wrapper

  describe('Debunks component', () => {
    beforeEach(() => {
      wrapper = shallow(<Debunks />)
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should iterate over the debunks array and return an array of image elements', () => {
      const mockState = {url: 'https://epic.gsfc.nasa.gov.png', id: 3, text: ''}
      const expected = [<img className='yessa' src={mockState.url}/>]
      expect(wrapper.find('.yessa')).toEqual(1)
    })
  })

  describe('mapStateToProps', () => {
    it('should return an object with the debunks array', () => {
      const mockState = {
        debunks: [{url: 'https://epic.gsfc.nasa.gov.png', id: 3, text: ''}],
      }
      const expected = {
        debunks: [{url: 'https://epic.gsfc.nasa.gov.png', id: 3, text: ''}]
      }
      console.log(mapStateToProps)
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })

})