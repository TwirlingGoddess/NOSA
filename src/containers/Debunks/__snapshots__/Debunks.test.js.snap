// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debunks container Debunks component should match the snapshot 1`] = `undefined`;
